pkg/scheduler/objects/preemption.go

```diff
@@ -698,6 +698,12 @@ func (qps *QueuePreemptionSnapshot) IsAtOrAboveGuaranteedResource() bool {
 	if qps == nil {
 		return false
 	}
+
+	// check the parent, as violations at any level mean we are not within limits
+	if qps.Parent.IsAtOrAboveGuaranteedResource() {
+		return true
+	}
+
```

Comment:
This and line 755 are behavioral changes. The JIRA says the type is "Test", but then that's a mistake, isn't it?